### DISCUSSION TITLE
Measuring open source framework adoption?

### DISCUSSION THREAD

**Discussion by alexcg1** [todogroup/ospology#33] [OSPO Discussions]
Hey all,

Posting [my Slack question](https://thetodogroup.slack.com/archives/C0GBYUWCB/p1636020022055200) here as well.

We're building an open source backend framework that lets people build their own search engines. Now we want to better track adoption. And by adoption we mean usage of our framework in professional projects.

We're currently tracking stuff like pip installs, Docker pulls, git clones, etc, etc. But actual adoption is difficult, especially since most professional use of our project would be for commercial projects which would overwhelmingly be closed source. (which is fully OK by our Apache-2 license)

We had a discussion sometime ago about telemetry and stuff like that, but we overwhelmingly decided against it. So that's not really an option.

Would love to hear thoughts and feedback!

**Comment by jsmanrique** [comment]
This is a common question for many open source project managers, and from time to time I wonder myself if it really matters, or more precisely "why does it matter" to you?

In my ideal OSS world, people develop OSS solutions to solve their own itches, so while it solves your own itches, does it really matter how many other people are using it?

If the component is used as part of a market solution, then, what really matters is the adoption of the solution itself.

But, in some cases, OSS tools are used as "marketing channels" to engage developers/users as future customers of side solutions or extra services. In that case, the component itself is an asset and the usual "marketing metrics" are put in place (downloads, comments, etc.). Some ideas that might be interesting to explore to get "adoption" might be:
* optional telemetry (several projects have used it), but always taking privacy seriously
* active users asking questions in forums or submitting issues in the issue tracking systems. This might be tricky because we will be biased by the "complaining users", not the satisfied ones (Some sort of [Survivorship bias](https://en.wikipedia.org/wiki/Survivorship_bias)
* surveis
